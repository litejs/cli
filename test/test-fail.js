
describe
.it("should fail on timeout", function(assert) {
	assert.setTimeout(5)
})
.it("should fail on multiple ends", function(assert) {
	setTimeout(this.end, 6)
	assert.end()
})
.it("should fail on assertion after end", function(assert) {
	assert.end()
	assert.ok(1)
})
.it("should fail on invalid assertions", function() {
	this.equal(
		[1, 2]  // <-- notice a comma is missing after "]"
		[3]
	)
	.equal(1, 2)
	.ok(0)
	.throws(function(){})
	.end()
})


