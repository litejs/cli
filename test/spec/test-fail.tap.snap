TAP version 13
# Unnamed TestSuite
not ok 1 - it should fail on timeout [0/0]
ok 2 - it should fail on multiple ends [0/0]
ok 3 - it should fail on assertion after end [0/0]
not ok 4 - it should fail on invalid assertions [0/4]
---
not ok 4 - it should fail on invalid assertions [0/4]
AssertionError#1: equal
expected: undefined
actual:   undefined
    at Object.<anonymous> (test/test-fail.js:15:7)
    at nextCase (test.js:324:14)
    at Object.endCase [as end] (test.js:367:23)
AssertionError#2: equal
expected: 2
actual:   1
    at Object.<anonymous> (test/test-fail.js:19:3)
    at nextCase (test.js:324:14)
    at Object.endCase [as end] (test.js:367:23)
AssertionError#3: 0
    at Object.<anonymous> (test/test-fail.js:20:3)
    at nextCase (test.js:324:14)
    at Object.endCase [as end] (test.js:367:23)
AssertionError#4: throws
    at Object.<anonymous> (test/test-fail.js:21:3)
    at nextCase (test.js:324:14)
    at Object.endCase [as end] (test.js:367:23)
---
not ok 1 - it should fail on timeout [0/0]
TIMEOUT: 5ms
...
# FAIL  tests 1, 4
1..4
# pass  2/4 [0/4]
---
not ok 3 - it should fail on assertion after end [0/1]
Error: assertion after end
...
# FAIL  tests 3
1..4
# pass  1/4 [0/4]
---
not ok 2 - it should fail on multiple ends [0/0]
Error: ended multiple times
...
# FAIL  tests 2
1..4
# pass  0/4 [0/4]
